### PR TITLE
docs: add References through BibTeX

### PR DESCRIPTION
**Pull request by umarcor**
This PR is based on #366.

Add a BibTeX file (`refs.bib`) and use it in the documentation through extension `sphinxcontrib-bibtex`.

### CODE REVIEW

**Comment by proppy** [comment]
maybe you could split the change from #366 so that it can be merged independently (it doesn't seems to be affecting the same files).

**Comment by umarcor** [comment]
@proppy done!

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,20 +1,13 @@
-git+https://github.com/SymbiFlow/sphinx_symbiflow_theme.git#egg=sphinx-symbiflow-theme
```
Comment (mithro) [inline docs/requirements.txt:1]: Seems to be a lot of unrelated changes to this file?

**Review by mithro — APPROVED** [review]
LGTM, just one minor comment!

File: docs/requirements.txt (hunk 1 above)
Comment (umarcor) [reply docs/requirements.txt]: Fixed!
